Healthy/Classes/Modules/Splash/DemoViewController.swift

```diff
@@ -0,0 +1,225 @@
+        skipButtonTopAnchor = skipButton.topAnchor.constraint(equalToSystemSpacingBelow: view.safeAreaLayoutGuide.topAnchor, multiplier: 2)
+        nextButtonTopAnchor = nextButton.topAnchor.constraint(equalToSystemSpacingBelow: view.safeAreaLayoutGuide.topAnchor, multiplier: 2)
+        
+        pageControlBottomAnchor?.isActive = true
+        skipButtonTopAnchor?.isActive = true
+        nextButtonTopAnchor?.isActive = true
+    }
+}
+
+// MARK: - DataSource
+
+extension DemoViewController: UIPageViewControllerDataSource {
+    
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
+
+        guard let currentIndex = pages.firstIndex(of: viewController) else { return nil }
+        
+        if currentIndex == 0 {
+            return pages.last               // wrap last
+        } else {
+            return pages[currentIndex - 1]  // go previous
+        }
+    }
+        
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerAfter viewController: UIViewController) -> UIViewController? {
+        
+        guard let currentIndex = pages.firstIndex(of: viewController) else { return nil }
+
+        if currentIndex < pages.count - 1 {
+            return pages[currentIndex + 1]  // go next
+        } else {
+            return pages.first              // wrap first
+        }
+    }
+}
+
+// MARK: - Delegates
+
+extension DemoViewController: UIPageViewControllerDelegate {
+    
+    // How we keep our pageControl in sync with viewControllers
+    func pageViewController(_ pageViewController: UIPageViewController, didFinishAnimating finished: Bool, previousViewControllers: [UIViewController], transitionCompleted completed: Bool) {
+        
+        guard let viewControllers = pageViewController.viewControllers else { return }
+        guard let currentIndex = pages.firstIndex(of: viewControllers[0]) else { return }
+        
+        pageControl.currentPage = currentIndex
+        animateControlsIfNeeded()
+    }
+    
+    private func animateControlsIfNeeded() {
+        let lastPage = pageControl.currentPage == pages.count - 1
+        
+        if lastPage {
+            hideControls()
+        } else {
+            showControls()
+        }
+
+        UIViewPropertyAnimator.runningPropertyAnimator(withDuration: 0.5, delay: 0, options: [.curveEaseOut], animations: {
+            self.view.layoutIfNeeded()
+        }, completion: nil)
+    }
+    
+    private func hideControls() {
+        pageControlBottomAnchor?.constant = -80
+        skipButtonTopAnchor?.constant = -80
+        nextButtonTopAnchor?.constant = -80
+    }
+
+    private func showControls() {
+        pageControlBottomAnchor?.constant = 16
+        skipButtonTopAnchor?.constant = 16
+        nextButtonTopAnchor?.constant = 16
```

Comment:
move any constants to a constant file & try to avoid much use of NSLayoutConstraint to avoid conflicts in auto layout at different screens